api/v1alpha1/basic_auth_types.go

```diff
@@ -0,0 +1,23 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import gwapiv1b1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+// BasicAuth defines the configuration for 	the HTTP Basic Authentication.
+type BasicAuth struct {
+	// The Kubernetes secret which contains the username-password pairs in
+	// htpasswd format, used to verify user credentials in the "Authorization"
+	// header.
+	//
+	// This is an Opaque secret. The username-password pairs should be stored in
```

Comment:
maybe also add a line mentioning that secret must live in the same namespace as the policy